packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -415,23 +443,33 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
-    }
+    const pathMatchConfig = this.match?.path?.bind(scope);
 
+    // Set prefix path match to '/' if none of path matches are defined.
+    const prefixPathMatch = pathMatchConfig ? pathMatchConfig.prefixPathMatch : '/';
+    const wholePathMatch = pathMatchConfig?.wholePathMatch;
     const headers = this.match?.headers;
-    validateMatchArrayLength(headers);
+    const queryParameters = this.match?.queryParameters;
+
+    if (prefixPathMatch && prefixPathMatch[0] !== '/') {
+      throw new Error(`Prefix Path for the match must start with \'/\', got: ${prefixPathMatch}`);
+    }
+    if (wholePathMatch?.exact && wholePathMatch.exact[0] !== '/') {
+      throw new Error(`Exact Path for the match must start with \'/\', got: ${wholePathMatch.exact}`);
+    }
```

Comment:
Can we actually move this validation to the `HttpRoutePatchMatch` subclasses instead? I think this will greatly simplify this class.

Reply:
Moved.